projects/stm32_F411RE_nucleo_64_mqtt_mutual_auth_demo/Core/Src/app_main.c

```diff
@@ -0,0 +1,201 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ * 1 tab == 4 spaces!
+ */
+
+#include <stdbool.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include "demo_config.h"
+
+/*-----------------------------------------------------------*/
+
+/* Unmark this to run comm interface deom. */
+// #define CELLULAR_COMM_IF_DEMO_TASK
+
+/*-----------------------------------------------------------*/
+
+extern bool setupCellular( void );
+
+extern void RunMQTTTask( void * pvParameters );
+
+UBaseType_t uxRand( void );
+
+/* The task function to setup cellular with thread ready environment. */
+static void CellularDemoTask( void * pvParameters );
+
+/*-----------------------------------------------------------*/
+
+void app_main( void )
+{
+    /* FreeRTOS Cellular Library init needs thread ready environment.
+     * CellularDemoTask invoke setupCellular to init FreeRTOS Cellular Library and register network.
+     * Then it runs the MQTT demo. */
+    xTaskCreate( CellularDemoTask,         /* Function that implements the task. */
+                 "CellularDemo",           /* Text name for the task - only used for debugging. */
+				 democonfigDEMO_STACKSIZE, /* Size of stack (in words, not bytes) to allocate for the task. */
+                 NULL,                     /* Task parameter - not used in this case. */
+				 democonfigDEMO_PRIORITY,  /* Task priority, must be between 0 and configMAX_PRIORITIES - 1. */
+                 NULL );     
+
+    vTaskStartScheduler();
+    for( ; ; );
+}
+/*-----------------------------------------------------------*/
+
+#ifdef CELLULAR_COMM_IF_DEMO_TASK
+    extern void CommIfDemo( void * pvParameters );
+    static void CellularDemoTask( void * pvParameters )
+    {
+        configPRINTF( ( "Comm interface demo.\r\n" ) );
+        CommIfDemo( pvParameters );
```

Comment:
How is CellularCommInterface initialized in this demo? Do we need to call setupCellular to init that?

Reply:
We can directly exercise the comm interface to communicate with cellular modem without cellular interface library. Cellular interface library provides help to handle AT commands.